test/units/galaxy/test_collection.py

```diff
@@ -0,0 +1,310 @@
+    os.makedirs(roles_target_tasks)
+    with open(os.path.join(roles_target_tasks, 'main.yml'), 'w+') as tasks_main:
+        tasks_main.write("---\n- hosts: localhost\n  tasks:\n  - ping:")
+        tasks_main.flush()
+
+    os.symlink(roles_target, roles_link)
+
+    actual = collection._build_files_manifest(input_dir)
+
+    linked_entries = [e for e in actual['files'] if e['name'].startswith('playbooks/roles/linked')]
+    assert len(linked_entries) == 3
+    assert linked_entries[0]['name'] == 'playbooks/roles/linked'
+    assert linked_entries[0]['ftype'] == 'dir'
+    assert linked_entries[1]['name'] == 'playbooks/roles/linked/tasks'
+    assert linked_entries[1]['ftype'] == 'dir'
+    assert linked_entries[2]['name'] == 'playbooks/roles/linked/tasks/main.yml'
+    assert linked_entries[2]['ftype'] == 'file'
+    assert linked_entries[2]['chksum_sha256'] == '9c97a1633c51796999284c62236b8d5462903664640079b80c37bf50080fcbc3'
+
+
+def test_build_with_symlink_inside_collection(collection_input):
+    input_dir, output_dir = collection_input
+
+    os.makedirs(os.path.join(input_dir, 'playbooks', 'roles'))
+    roles_link = os.path.join(input_dir, 'playbooks', 'roles', 'linked')
+
+    roles_target = os.path.join(input_dir, 'roles', 'linked')
+    roles_target_tasks = os.path.join(roles_target, 'tasks')
+    os.makedirs(roles_target_tasks)
+    with open(os.path.join(roles_target_tasks, 'main.yml'), 'w+') as tasks_main:
+        tasks_main.write("---\n- hosts: localhost\n  tasks:\n  - ping:")
+        tasks_main.flush()
+
+    os.symlink(roles_target, roles_link)
+
+    collection.build_collection(input_dir, output_dir, False)
+
+    output_artifact = os.path.join(output_dir, 'ansible_namespace-collection-0.1.0.tar.gz')
+    assert tarfile.is_tarfile(output_artifact)
+
+    actual = tarfile.open(output_artifact)
+    try:
+        members = actual.getmembers()
+
+        linked_members = [m for m in members if m.path.startswith('playbooks/roles/linked/tasks')]
+        assert len(linked_members) == 2
+        assert linked_members[0].name == 'playbooks/roles/linked/tasks'
+        assert linked_members[0].isdir()
+
+        assert linked_members[1].name == 'playbooks/roles/linked/tasks/main.yml'
+        assert linked_members[1].isreg()
+
+        linked_task = actual.extractfile(linked_members[1].name)
+        actual_task = secure_hash_s(linked_task.read())
+        linked_task.close()
+
+        assert actual_task == 'f4dcc52576b6c2cd8ac2832c52493881c4e54226'
+    finally:
+        actual.close()
+
+
+def test_publish_missing_file():
+    expected = "The collection path specified '/fake/path' does not exist."
+
+    with pytest.raises(AnsibleError, match=expected):
+        collection.publish_collection('/fake/path', None, None, False, True)
+
+
+def test_publish_not_a_tarball():
+    expected = "The collection path specified '{0}' is not a tarball, use 'ansible-galaxy collection build' to " \
```

Comment:
I recommend using braces instead of escaping EOL.
```suggestion
    expected = (
    	"The collection path specified '{0}' is not a tarball, "
    	"use 'ansible-galaxy collection build' to "
        "create a proper release artifact."
    )
```

Reply:
Personal preference really, I'm more of a fan of the other direction.

Reply:
I see. I usually don't want that trailing ` \` to be accidentally removed which can later lead to the next line being readjusted creating a perfectly valid string literal part not assigned to anything (like docstrings).
Also, shorter lines improve readability in general, allowing readers to stop jumping back and forth with their eyes and just read the thing vertically w/o losing the context.

Reply:
> Also, shorter lines improve readability in general, allowing readers to stop jumping back and forth with their eyes and just read the thing vertically w/o losing the context.

Isn't that why we have a line length of 120 chars :)